projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js

```diff
@@ -31,27 +45,29 @@ export default function SubscribePanels() {
 		} );
 	}, [] );
 
-	// Only show this for posts for now (subscriptions are only available on posts).
-	const postType = useSelect( select => select( editorStore ).getCurrentPostType(), [] );
 	const postWasEverPublished = useSelect(
 		select =>
 			select( editorStore ).getEditedPostAttribute( 'meta' )?.jetpack_post_was_ever_published,
 		[]
 	);
 
-	if ( 'post' !== postType ) {
+	// Subscriptions will not be triggered on private sites (on WordPress.com simple and WoA),
+	// nor on sites that have not been launched yet.
+	if ( isPrivateSite() || isComingSoon() ) {
 		return null;
 	}
 
-	// Subscriptions will not be triggered for a post that was already published in the past
-	if ( postWasEverPublished ) {
+	// Subscriptions are only available for posts. Additionally, we will allow access level selector for pages.
+	// TODO: Make it available for pages later.
+	if ( postType !== 'post' ) {
 		return null;
 	}
 
-	// Subscriptions will not be triggered on private sites (on WordPress.com simple and WoA),
-	// nor on sites that have not been launched yet.
-	if ( isPrivateSite() || isComingSoon() ) {
-		return null;
+	// Subscriptions will not be triggered for a post that was already published in the past and the email was sent.
+	// We still need to render the access level selector, as historical posts need a way to edit their access level for people visiting them on the web.
+	// TODO: Additionally, pages also can be protected. They will not send an email, but can be a resource that needs the acces selector.
+	if ( postWasEverPublished ) {
+		return renderAccessLevelSelectorPanel( setPostMeta, accessLevel );
```

Comment:
Same. I think it would be best to use the tag version...

Reply:
Maybe we could actually use that check directly isn the last "return" and prevent some complexity?

Reply:
> Maybe we could actually use that check directly isn the last "return" and prevent some complexity?

I am not sure I follow

Reply:
See https://github.com/Automattic/jetpack/pull/29247#discussion_r1124110423
But again, it doesn't make any difference.